File: vdp-node/test/visacheckout/getpaymentdatatest.js
var request = require('request');
var xPayUtil = require('../../libs/xpayutil.js');
var config = require('../../config/configuration.json');
var expect = require('chai').expect;
var req = request.defaults();
var randomstring = require('randomstring');

describe('Get Payment Data', function() {

	it('Get Payment Information Test',function(done){
		this.timeout(10000);
		var apiKey = config.apiKey;
		var baseUri = 'wallet-services-web/';
		var resourcePath = 'payment/data/{callId}';
		resourcePath = resourcePath.replace('{callId}', config.checkoutCallId);
		var queryParams = 'apikey=' + apiKey;
		console.log("URL :" + config.visaUrl + baseUri + resourcePath + '?' + queryParams);
		req.get({
			uri : config.visaUrl + baseUri + resourcePath + '?' + queryParams,
			headers: {
				'content-type' : 'application/json',
				'x-pay-token' : xPayUtil.getXPayToken(resourcePath, queryParams, ''),
				'x-correlation-id' : randomstring.generate({length:12, charset: 'alphanumeric'}) + '_SC'
			}
		}, function(error, response, body) {
			if (!error) {
				console.log("Response Code: " + response.statusCode);
				console.log("Headers:");
				for(var item in response.headers) {
					console.log(item + ": " + response.headers[item]);
				}
				console.log("Body: "+ JSON.stringify(JSON.parse(body), null, 4));
				assert.equal(response.statusCode, 200);
			} else {
				console.log(error);
				assert(false);
			}
			done();
		}
		);
	});
});

File: vdp-ruby/test/visacheckout/get_payment_data_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/x_pay_utils', __FILE__)

class GetPaymentDataTest < Test::Unit::TestCase
  
  def setup
    @config = YAML.load_file('configuration.yml')
    @x_pay_util = XPayUtils.new
  end
  
  def test_getPaymentInfo
      base_uri = 'wallet-services-web/'
      resource_path = 'payment/data/{callId}'
      resource_path = resource_path.sub('{callId}',@config['checkoutCallId'])
      api_key = @config['apiKey']
      url = "#{@config['visaUrl']}#{base_uri}#{resource_path}?apikey=#{api_key}"
      shared_secret = @config['sharedSecret']
      query_string = "apikey=#{api_key}"
      xpay_token = @x_pay_util.get_xpay_token(resource_path,query_string,'')
      puts "Get Payment Information Test"
      puts url
      # Passing correlation id header (x-correlation-id) is optional while making API calls.  
      correlation_id = (0...12).map { (48 + rand(10)).chr }.join
      headers = {'x-pay-token'=> xpay_token, 'accept' => 'application/json', 'x-correlation-id'=> "#{correlation_id}_SC"}
      begin
        response = RestClient::Request.execute(
        :method => :get,
        :url => url,
        :headers => headers
        )
      puts "Response Status : #{response.code.to_s}"
      puts "Response Headers : " 
      for header,value in response.headers
      puts "#{header.to_s} : #{value.to_s}"
      end
      puts "Response Body : " + JSON.pretty_generate(JSON.parse(response.body))
      rescue RestClient::ExceptionWithResponse => e
        puts "Response Status : #{e.response.code.to_s}"
        puts "Response Headers : " 
        for header,value in e.response.headers
        puts "#{header.to_s} : #{value.to_s}"
        end
        puts "Response Body : " + JSON.pretty_generate(JSON.parse(e.response.body))
        assert(false, "Get Payment Information test failed")
      end
    end
end